### DISCUSSION TITLE
Проверка\Предупреждение для offloading

### DISCUSSION THREAD

**Discussion by numbereleven-a** [bol-van/zapret#591] [Ideas]
Часто вижу что ты на форуме, когда у людей проблема, спрашиваешь не включен ли offloading.
Может быть есть смысл добавить проверку включен ли он при блокчеке например? 
Ну или вывести предупреждение о необходимости его отключить?

**Comment by bol-van** [comment]
Блокчек делался с минимумом зависимостей от дистрибутива и системы.
Он работает на всем, даже на pfsense и openbsd.
offloading реализуется средствами nftables на новых ядрах в openwrt и традиционном linux
в openwrt есть отдельная установка в UCI конфиге, которая заставляет fw4 добавлять в nftables цепочки для включения безусловного оффлоада tcp и udp
в традиционных системах никаких установок нет, все определяется лишь ruleset nftables
в старых openwrt оффлоад включается через проприетарный для openwrt target FLOWOFFLOAD. UCI настройка тоже есть.

то есть наличие оффлоада можно однозначно распознать только по наличию цепочки в ip/nf tables.
но она может быть условной. или вообще не задействоваться.
не хочется это все парсить и додумывать.

а еще, кстати, на всяких стоках это может быть отсебятина производителя, которая включается и отключается через проприетарные интерфейсы. что там на кинетике я вообще не представляю
